scio-jdbc/src/main/scala/com/spotify/scio/jdbc/package.scala

```diff
@@ -84,14 +85,31 @@ package object jdbc {
     def apply[T](jdbcIoOptions: JdbcIoOptions): JdbcIO[T] = JdbcIO[T](jdbcIoId(jdbcIoOptions))
   }
 
-  private def jdbcIoId(opts: JdbcConnectionOptions, query: String): String =
-    s"${opts.username}:${opts.password}@${opts.connectionUrl}:$query"
+  private def jdbcIoId(opts: JdbcConnectionOptions, query: String): String = {
+    val user = if (opts.password.isDefined) {
+      s"${opts.username}:${opts.password}"
```

Comment:
Should this be `opts.password.get`? You could also do `val user = opts.password.fold(s"${opts.username}")(pass => s"${opts.username}:${opts.password}")` which is more idiomatic Scala.

Reply:
yes - i should also add a test for that.